src/hooks/PersistedQueryProvider.tsx

```diff
@@ -0,0 +1,58 @@
+import { createAsyncStoragePersister } from '@tanstack/query-async-storage-persister';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { PersistQueryClientProvider } from '@tanstack/react-query-persist-client';
+import { isArray } from 'lodash';
+import React from 'react';
+import { Query, QueryClient } from '@tanstack/query-core';
+
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      cacheTime: 1000 * 60 * 60 * 24, // 24 hours
+    },
+  },
+});
+
+const asyncStoragePersister = createAsyncStoragePersister({
+  storage: AsyncStorage,
+});
+
+const queriesToPersist = ['profile'];
+
+const filterQueryByKey = (q: any) => {
+  // PersistClientCore imports a legacy version of type Query
+  const query = q as Query;
+  if (isArray(query.queryKey)) {
+    // Match on first part of key only to ignore unique ids
+    const keyPart = query.queryKey?.[0];
+    if (queriesToPersist.includes(keyPart)) {
+      return true;
+    }
+  }
+
+  return false;
+};
```

Comment:
Nice. Just confirming: this will only persist keys that look like `['profile', ...]`, right?

Reply:
Yep, that is the case for the time being. I really didn't want to suddenly adopt the default behavior of persisting everything, that seemed like it would be problematic.